google-cloud-spanner/src/main/java/com/google/cloud/spanner/Statement.java

```diff
@@ -246,4 +248,56 @@ StringBuilder toString(StringBuilder b) {
     }
     return b;
   }
+
+  public static final class StatementFactory {
+    private final Dialect dialect;
+
+    StatementFactory(Dialect dialect) {
+      this.dialect = dialect;
+    }
+
+    public Statement of(String sql) {
+      return Statement.of(sql);
+    }
+
+    /**
+     * @param sql SQL statement with unnamed parameter denoted as ?
+     * @param values list of values which needs to replace ? in the sql
+     * @return Statement object
+     *     <p>This function accepts the SQL statement with unnamed parameters(?) and accepts the
+     *     list of objects to replace unnamed parameters. Primitive types are supported
+     *     <p>For Date column, following types are supported
+     *     <ul>
+     *       <li>java.util.Date
+     *       <li>LocalDate
+     *       <li>com.google.cloud.Date
+     *     </ul>
+     *     <p>For Timestamp column, following types are supported. All the dates should be in UTC
+     *     format. Incase if the timezone is not in UTC, spanner client will convert that to UTC
+     *     automatically
+     *     <ul>
+     *       <li>LocalDateTime
+     *       <li>OffsetDateTime
+     *       <li>ZonedDateTime
+     *     </ul>
+     *     <p>
+     * @see DatabaseClient#newStatementFactory
+     */
+    public Statement of(String sql, Object... values) {
```

Comment:
I think that it would be good to clearly call out what the purpose is of this method, also in the method name.

```suggestion
    public Statement withUnnamedParameters(String sql, Object... values) {
```